oak_runtime/src/introspect.rs

```diff
@@ -111,20 +111,47 @@ fn find_client_file(path: &str) -> Option<(Vec<u8>, String)> {
         .as_str();
 
     match subpath {
-        "/index.js" => Some((
-            include_bytes!("../introspection_browser_client/dist/index.js.gz").to_vec(),
-            "application/javascript".to_string(),
-        )),
-        "/graphvizlib.wasm" => Some((
-            include_bytes!("../introspection_browser_client/dist/graphvizlib.wasm.gz").to_vec(),
-            "application/wasm".to_string(),
-        )),
+        "/index.js" => {
+            let mut response = Response::new(Body::from(
+                include_bytes!("../introspection_browser_client/dist/index.js.gz").to_vec(),
+            ));
+            let headers = response.headers_mut();
+            headers.insert(CONTENT_TYPE, "application/javascript".parse().unwrap());
+            headers.insert(hyper::header::CONTENT_ENCODING, "gzip".parse().unwrap());
+
+            Some(response)
+        }
+        "/graphvizlib.wasm" => {
+            let mut response = Response::new(Body::from(
+                include_bytes!("../introspection_browser_client/dist/graphvizlib.wasm.gz").to_vec(),
+            ));
+            let headers = response.headers_mut();
+            headers.insert(CONTENT_TYPE, "application/wasm".parse().unwrap());
+            headers.insert(hyper::header::CONTENT_ENCODING, "gzip".parse().unwrap());
```

Comment:
An odd Rust thing happened here: I had to pick the `CONTENT_ENCODING` inline, instead importing it at the top of the file to fix a rust warning:

![image](https://user-images.githubusercontent.com/8875406/95190568-6f2c6980-07c7-11eb-8e61-17cae401e283.png)

I suspect the reason is that not all items in the match use this type? It seemed like weird behaviour.

Reply:
I think that's just because the import is only used if `oak_introspection_client` is on, but it is imported unconditionally. Maybe it would make sense to have a submod that gets excluded completely if the feature is not enabled, and would contain also the imports? Otherwise you would have to feature gate the imports too, or use them inline as you did here.

Reply:
Ah that explains it. :) Initially I ruled that possibility out since `CONTENT_TYPE` works fine, but indeed that type is used outside of the feature flagged function. 

Will implement your suggestion of the conditional import.